src/Motor.Extensions.Hosting.Kafka/KafkaMessageConsumer.cs

```diff
@@ -75,31 +80,43 @@ public async Task ExecuteAsync(CancellationToken token = default)
     {
         await Task.Run(async () =>
         {
-            while (!token.IsCancellationRequested)
+            var committer = ExecuteCommitLoopAsync(token);
+
+            try
             {
-                await _messageSemaphore.WaitAsync(token);
-                try
+                while (!token.IsCancellationRequested)
                 {
-                    var msg = _consumer?.Consume(token);
-                    if (msg is { IsPartitionEOF: false })
+                    try
                     {
-                        SingleMessageHandlingAsync(msg, token);
+                        if (!await _processedMessages.Writer.WaitToWriteAsync(token))
+                        {
+                            break;
+                        }
+
+                        var msg = _consumer?.Consume(token);
+                        if (msg is { IsPartitionEOF: false })
+                        {
+                            await _processedMessages.Writer.WriteAsync(SingleMessageHandlingAsync(msg, token), token);
+                        }
+                        else
+                        {
+                            _logger.LogDebug(LogEvents.NoMessageReceived, "No messages received");
+                        }
                     }
-                    else
+                    catch (Exception e) when (e is not OperationCanceledException or ChannelClosedException)
                     {
-                        _logger.LogDebug(LogEvents.NoMessageReceived, "No messages received");
+                        _logger.LogError(LogEvents.MessageReceivedFailure, e, "Failed to receive message.");
                     }
                 }
-                catch (OperationCanceledException)
-                {
-                    _logger.LogInformation(LogEvents.TerminatingKafkaListener, "Terminating Kafka listener...");
-                    break;
-                }
-                catch (Exception e)
-                {
-                    _logger.LogError(LogEvents.MessageReceivedFailure, e, "Failed to receive message.");
-                }
+
+                await committer;
             }
+            catch (Exception e) when (e is OperationCanceledException or ChannelClosedException)
+            {
```

Comment:
Codacy expects at least a comment inside of an otherwise empty catch block

Reply:
Part of next commit.